src/Google.Logging.Log4Net/GoogleStackdriverAppender.cs

```diff
@@ -0,0 +1,274 @@
+                DelayMultiplier = ServerErrorBackoffMultiplier,
+                MaxDelay = TimeSpan.FromSeconds(ServerErrorBackoffMaxDelaySeconds),
+            };
+            _logUploader = new LogUploader(
+                _client, _scheduler, _clock,
+                _logQ, logsLostWarningEntry, MaxUploadBatchSize,
+                serverErrorBackoffSettings);
+        }
+
+        private async Task InitNextId()
+        {
+            _nextId = await _logQ.GetNextIdAsync();
+        }
+
+        private LogEntry BuildLogEntry(LoggingEvent loggingEvent)
+        {
+            var labels = new Dictionary<string, string>();
+            if (_withMetaData.Contains(MetaDataType.Location))
+            {
+                labels.Add(nameof(MetaDataType.Location) + ".FileName", loggingEvent.LocationInformation.FileName);
+                labels.Add(nameof(MetaDataType.Location) + ".ClassName", loggingEvent.LocationInformation.ClassName);
+                labels.Add(nameof(MetaDataType.Location) + ".LineNumber", loggingEvent.LocationInformation.LineNumber);
+            }
+            if (_withMetaData.Contains(MetaDataType.Identity))
+            {
+                labels.Add(nameof(MetaDataType.Identity), loggingEvent.Identity);
+            }
+            if (_withMetaData.Contains(MetaDataType.ThreadName))
+            {
+                labels.Add(nameof(MetaDataType.ThreadName), loggingEvent.ThreadName);
+            }
+            if (_withMetaData.Contains(MetaDataType.UserName))
+            {
+                labels.Add(nameof(MetaDataType.UserName), loggingEvent.UserName);
+            }
+            if (_withMetaData.Contains(MetaDataType.Domain))
+            {
+                labels.Add(nameof(MetaDataType.Domain), loggingEvent.Domain);
+            }
+            if (_withMetaData.Contains(MetaDataType.LoggerName))
+            {
+                labels.Add(nameof(MetaDataType.LoggerName), loggingEvent.LoggerName);
+            }
+            if (_withMetaData.Contains(MetaDataType.Level))
+            {
+                labels.Add(nameof(MetaDataType.Level), loggingEvent.Level.Name);
+            }
+            foreach (var customLabel in _customLabels)
+            {
+                labels.Add(customLabel.Key, customLabel.Value);
+            }
+            return new LogEntry
+            {
+                TextPayload = RenderLoggingEvent(loggingEvent),
+                Severity = s_levelMap[loggingEvent.Level],
+                Timestamp = loggingEvent.TimeStamp.ToTimestamp(),
+                LogName = _logName,
+                Resource = _resource,
+                Labels = { labels },
+            };
+        }
+
+        private void Write(IEnumerable<LogEntry> logEntries)
+        {
+            DateTimeRange logEntriesLost;
+            lock (_enqueueLock)
+            {
+                if (_initNextIdTask != null)
+                {
+                    _initNextIdTask.Wait();
```

Comment:
Task.Run() added above, which should prevent deadline. If I'm wrong, please say!